src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -399,6 +399,29 @@ public static String formatActivityLogTime(Instant instant, ZoneId adminTimeZone
         return formatInstant(instant, adminTimeZone, "MM/dd/yyyy HH:mm:ss SSS");
     }
 
+    /**
+     * Formatter for timezone offsets of type double.
+     * Example: a offset of -8.5 returns "UTC -08:30".
+     * Precision can be guaranteed to a quarter of an hour (covered by tests).
+     *
+     * @param hourOffsetTimeZone the hour offset between -19 to 19 non inclusive.
+     * @return UTC offset string.
+     * @deprecated Timezones should not be passed around as doubles, use ZoneId/ZoneOffset instead.
+     */
+    @Deprecated
+    public static String formatTimeZoneToUtcOffset(double hourOffsetTimeZone) {
+
+        int hours = (int) hourOffsetTimeZone;
+        int minutes = (int) ((hourOffsetTimeZone - hours) * 60.0);
+
+        ZoneOffset offset = ZoneOffset.ofHoursMinutes(hours, minutes);
+
+        if ("Z".equals(offset.toString())) {
```

Comment:
The guard clause can be the very first statement, even before converting to seconds?

Reply:
Oh wait, I stashed that commit fixing that and forgot to apply it, hang on.

Reply:
Yep but I remember your new version does the conversion to seconds first, which is wasted computation.

Reply:
We can't check if `hourOffsetTimeZone` is equal to 0 directly because it's a double and it'll never return true even if `hourOffsetTimeZone` is passed in as 0.0, unless you go the hacky way and do something like `Math.abs(hourOffsetTimeZone) < 2 * Double.MIN_VALUE`. We can't cast it to `int` either because there's a possibility that `hourOffsetTimeZone` is something like 0.5 which would get rounded down to 0 during casting. I think computing the offset in seconds is a sensible way to deal with this.

Reply:
Hmm, why are we moving the function from `StringHelper` to here and discussing so much about it? It's deprecated anyway so shouldn't we just mark it as such in `StringHelper` and then remove it after we have converted timezone to `ZoneId`?

Reply:
> I remember having this drilled into my head to never compare doubles using ==.

Yea I had that instinct too, but I think those best practices are only necessary when doing arithmetic with doubles. We never perform arithmetic on the `timeZone` field, so a zero here should be an exact zero.

Reply:
> Hmm, why are we moving the function from StringHelper to here and discussing so much about it? It's deprecated anyway so shouldn't we just mark it as such in StringHelper and then remove it after we have converted timezone to ZoneId?

You have a point there